### PR TITLE
fixing e2e tasks in grunt and gulp, adding e2e tests for user module

### LINKED ISSUE #929: E2E tests not executing
after running `gulp protractor`
```
Error: Angular could not be found on the page http://localhost:3000/authentication/signin : retries looking for angular exceeded
```
from the angular [docs](https://docs.angularjs.org/guide/e2e-testing):
> ## Caveats
> Protractor does not work out-of-the-box with apps that bootstrap manually using angular.bootstrap. You must use the ng-app directive.

It looks like we manually bootstrap [here](https://github.com/meanjs/mean/blob/master/modules/core/client/app/init.js#L71)
Do we need to add `np-app` somewhere?

### PR DESCRIPTION
**Pull request by jloveland**
fixes #929 and #934

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,7 +17,10 @@
   "scripts": {
     "start": "grunt",
     "test": "grunt test",
-    "postinstall": "bower install --config.interactive=false"
+    "postinstall": [
+      "bower install --config.interactive=false",
+      "./node_modules/protractor/bin/webdriver-manager update"
```
Comment (mleanos) [inline package.json]: Can this be added as a Grunt task that is coupled with the protractor task? I'm concerned that not all users will want to deal with protractor & the e2e tests at the time of package install.

Comment (mleanos) [inline package.json]: Something like this in the Grunt file would probably suffice.. 
https://github.com/meanjs/mean/blob/master/gulpfile.js#L198

**Comment by mleanos** [comment]
@jloveland I think there's a bit too much going on with this PR. I'm worried it will take some time to review. 

Would it be possible to just add the protractor task to the Grunt file, and fix the existing e2e tests? Perhaps in a separate PR; that way you can keep all this work here. With that approach we would get the bug fix for the Protractor tests out quickly, and then we could start building out the e2e tests.

File: package.json (hunk 1 above)
Comment (jloveland) [inline package.json]: What do you think about this?

https://www.npmjs.com/package/grunt-protractor-webdriver

Comment (mleanos) [inline package.json]: If we use that package, would we still be required to run the server with our protractor test task? I couldn't tell from the documentation.

Either way, `grunt-protractor-webdriver` looks promising. I'm gonna take a shot at testing it.

**Comment by jloveland** [comment]
@mleanos, I agree, I will separate into different PRs.

File: package.json (hunk 1 above)
Comment (jloveland) [inline package.json]: @mleanos I didn't like `grunt-protractor-webdriver` after testing it out..take a look at https://github.com/meanjs/mean/pull/939 for the addition of the webdriver update to the grunt file.

**Pull request closed without merging**